### PR TITLE
Add addon template api

### PR DESCRIPTION
**Pull request by zhujian7**

### CODE REVIEW

File: addon/v1alpha1/types_addontemplate.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+// +kubebuilder:printcolumn:name="ADDON NAME",type=string,JSONPath=`.spec.addonName`
+
+// AddOnTemplate is the Custom Resource object, it is used to describe
+// how to deploy the addon agent and how to register the addon.
+//
+// AddOnTemplate is a cluster-scoped resource, and will only be used
+// on the hub cluster.
+type AddOnTemplate struct {
```
Comment (qiujian16) [inline addon/v1alpha1/types_addontemplate.go:21]: ManagedClusterAddOnTemplate?

File: addon/v1alpha1/types_addontemplate.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	// +optional
+	HubPermissions []HubPermissionConfig `json:"hubPermissions,omitempty"`
+}
+
+// HubPermissionConfig configures the permission of the addon agent to access the hub cluster.
+// Will create a RoleBinding in the same namespace as the managedClusterAddon to bind the user
+// provided ClusterRole/Role to the "system:open-cluster-management:cluster:<cluster-name>:addon:<addon-name>"
+// Group.
+type HubPermissionConfig struct {
+	// Type of the permissions setting. It defines how to bind the roleRef
```
Comment (qiujian16) [inline addon/v1alpha1/types_addontemplate.go]: we need to describe the meaning of CurrentCluster and singlenamespace

**Review by qiujian16 — COMMENTED** [review]
/approve

two minor comments

File: addon/v1alpha1/types_addontemplate.go (hunk 2 above)
Comment (zhujian7) [reply addon/v1alpha1/types_addontemplate.go]: described [here](https://github.com/open-cluster-management-io/api/pull/239/files#diff-cc2eb9f759617220330a98f0609b906346b9f986ccaf027c4449db17fbf1f4dcR66-R72), should I move them into the `HubPermissionConfig` struct？

File: addon/v1alpha1/types_addontemplate.go (hunk 1 above)
Comment (zhujian7) [reply addon/v1alpha1/types_addontemplate.go:21]: should it start with "addon" to be consistent with addonDeploymentConfig?

Comment (qiujian16) [reply addon/v1alpha1/types_addontemplate.go:21]: hrm, that makes sense also

File: addon/v1alpha1/types_addontemplate.go (hunk 2 above)
Comment (qiujian16) [reply addon/v1alpha1/types_addontemplate.go]: I think we should for generating the openapi doc

Comment (zhujian7) [reply addon/v1alpha1/types_addontemplate.go]: description added

**Pull request merged**